components/providers/PolarisProvider.jsx

```diff
@@ -1,14 +1,10 @@
 import { useCallback } from "react";
 import { AppProvider } from "@shopify/polaris";
-import { useNavigate } from "@shopify/app-bridge-react";
 import "@shopify/polaris/build/esm/styles.css";
 import { getPolarisTranslations } from "../../utils/i18nUtils";
 
 function AppBridgeLink({ url, children, external, ...rest }) {
-  const navigate = useNavigate();
-  const handleClick = useCallback(() => {
-    navigate(url);
-  }, [url]);
+  const handleClick = useCallback(() => window.open(url), [url]);
```

Comment:
Considering we are fixing the template in this PR, can we wait til the back stack fix is out. I would like to bring `shopify.navigation.onNavigate` to the template to fix back stack as a green path. https://github.com/Shopify/app-bridge-next/pull/345/files#diff-930d0931e69b8c66af5536e0b6fa87053b0e051c7ab29c6cee9abe1f37fb5aa8R25

Reply:
Yes, we do. There are two options. I think option 1 is the best for most cases.

- Option 1: still call `navigate(url);` like before but need to call extra method like so
```js

  const handleClick = useCallback(async () => {
    await shopify.saveBar.leaveConfirmation();
    navigate(url);
  }, [url]);
```
- Option 2: subscribe to `shopify.navigation.onNavigate` like so
```
shopify.navigate.onNavigate(() => {
    const {url, intercept} = event;
    // call intercept if the url is a SPA
    intercept({
      async handler() {
        navigate(url);
      },
    });
});
...
const handleClick = useCallback(() => window.open(url), [url]);

```